bftengine/src/bftengine/KeyExchangeManager.cpp

```diff
@@ -113,6 +116,42 @@ void KeyExchangeManager::loadPublicKeys() {
   notifyRegistry();
 }
 
+void KeyExchangeManager::exchangeTlsKeys(const SeqNum& bft_sn) {
+  auto repId = bftEngine::ReplicaConfig::instance().replicaId;
+  auto keys = concord::util::crypto::Crypto::instance().generateECDSAKeyPair(
+      concord::util::crypto::KeyFormat::PemFormat, concord::util::crypto::CurveType::secp384r1);
+  auto cert = concord::util::crypto::Crypto::instance().generateSelfSignedCertificate(
+      keys, "node" + std::to_string(repId) + "ser", repId);
+  std::string pk_path =
+      bftEngine::ReplicaConfig::instance().certificatesRootPath + "/" + std::to_string(repId) + "/server/pk.pem";
+  concord::secretsmanager::SecretsManagerPlain psm_;
```

Comment:
We don't need to decide which secrets manager to use, we work with a provided one

Reply:
fixed